src/features/speedup/components/SpeedUpMonitor.tsx

```diff
@@ -0,0 +1,81 @@
+import { Alert, AlertTitle, Box, Button, SvgIcon, Typography } from '@mui/material'
+import { SpeedUpModal } from '@/features/speedup/components/SpeedUpModal'
+import Rocket from '@/public/images/common/rocket.svg'
+import { useCounter } from '@/components/common/Notifications/useCounter'
+import { useState } from 'react'
+import { PendingStatus, type PendingTx } from '@/store/pendingTxsSlice'
+import useAsync from '@/hooks/useAsync'
+import { isSmartContract, useWeb3ReadOnly } from '@/hooks/wallets/web3'
+import useWallet from '@/hooks/wallets/useWallet'
+import { sameAddress } from '@/utils/addresses'
+
+type SpeedUpMonitorProps = {
+  txId: string
+  pendingTx: PendingTx
+  modalTrigger: 'alertBox' | 'alertButton'
+}
+
+const SPEED_UP_THRESHOLD_IN_SECONDS = 15
+
+export const SpeedUpMonitor = ({ txId, pendingTx, modalTrigger = 'alertBox' }: SpeedUpMonitorProps) => {
+  const [openSpeedUpModal, setOpenSpeedUpModal] = useState(false)
+  const wallet = useWallet()
+  const counter = useCounter(pendingTx.submittedAt)
+  const web3ReadOnly = useWeb3ReadOnly()
+  const [transaction] = useAsync(async () => {
+    if (!pendingTx.txHash || !web3ReadOnly) return null
+    return web3ReadOnly?.getTransaction(pendingTx.txHash)
+  }, [pendingTx.txHash, web3ReadOnly])
+
+  const [smartContract] = useAsync(async () => {
+    if (!pendingTx.signerAddress || !web3ReadOnly) return false
+    return await isSmartContract(web3ReadOnly, pendingTx.signerAddress)
```

Comment:
```suggestion
    return isSmartContract(web3ReadOnly, pendingTx.signerAddress)
```

No need to await this